test/basicfuns.jl

```diff
@@ -44,6 +44,34 @@
     @test iszero(xlog1py(0 + im * 0, -1 + im * Inf))
 end
 
+@testset "xexpx" begin
+    for z in (0, 0.0, 0f0, false, -Inf)
+        @test iszero(@inferred xexpx(z))
+    end
+    for nan in (NaN16, NaN32, NaN64)
+        @test isnan(@inferred xexpx(nan))
+    end
+    for inf in (Inf16, Inf32, Inf64)
+        @test inf === @inferred xexpx(inf)
+    end
+    for u in (1, true, 1.0, 1f0)
+        @test exp(u) === @inferred xexpx(u)
+    end
+    @test 2exp(2) ≈ @inferred xexpx(2)
```

Comment:
```suggestion
    @test @inferred(xexpx(2)) ≈ 2 * exp(2)
```

Reply:
Alternatively, one could also test a random number:
```suggestion
    for x in (randn(), randn(Float32))
        @test @inferred(xexpx(x)) isa typeof(x)
        @test xexpx(x) ≈ x * exp(x)
    end
```